packages/vow/src/types.js

```diff
@@ -86,3 +86,17 @@ export {};
  * @property {(value: T, ...args: C) => Vow<TResult1> | PromiseVow<TResult1> | TResult1} [onFulfilled]
  * @property {(reason: any, ...args: C) => Vow<TResult2> | PromiseVow<TResult2> | TResult2} [onRejected]
  */
+
+/**
+ * Converts a vow or promise to a promise, ensuring proper handling of ephemeral promises.
+ *
+ * @template [T=any]
+ * @template [TResult1=T]
+ * @template [TResult2=never]
+ * @template {any[]} [C=any[]]
+ * @callback AsPromiseFunction
+ * @param {ERef<T | Vow<T>>} specimenP
+ * @param {Watcher<T, TResult1, TResult2, C>} [watcher]
+ * @param {C} [watcherArgs]
+ * @returns {Promise<TResult1 | TResult2>}
+ */
```

Comment:
This is identical to the type for `when`, minus the return line. We might consider extracting that logic and extending it, and/or using something like the PromiseToVow helper from @agoric/orchestration that should maybe live here instead: https://github.com/Agoric/agoric-sdk/blob/194ac48dcbc756242972eebf46567f3c1d9d0c15/packages/orchestration/src/internal.ts#L3-L16

Reply:
Shouldn't lines 14-15 above be removed since they're redundant?